core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -374,10 +375,16 @@ class ControllerApis(val requestChannel: RequestChannel,
         }
       }
     }
-    val authorizedTopicNames = if (hasClusterAuth) {
-      topicNames.asScala
-    } else {
-      getCreatableTopics.apply(topicNames.asScala)
+    val authorizedTopicNames = {
+      /* The cluster metatdata topic is an internal topic with a different implementation. The user should not be
+       * allowed to create it as a regular topic.
+       */
+      val allowedTopicNames = topicNames.asScala.diff(Set(Topic.CLUSTER_METADATA_TOPIC_NAME))
```

Comment:
Similar question to below -- should we log a warning if the user passed in `__cluster_metadata`?